conda_forge_tick/migrators/pip_check.py

```diff
@@ -0,0 +1,68 @@
+import typing
+from typing import Any
+from ruamel.yaml import YAML
+
+from conda_forge_tick.xonsh_utils import indir
+from conda_forge_tick.migrators.core import MiniMigrator
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict
+
+
+def _has_python_in_host(host):
+    host_set = {r.split(" ")[0] for r in host}
+    return bool(host_set & set(["python"])), host_set
+
+
+def _adjust_test_dict(test):
+    if 'requires' in test:
+        test['requires'].append('pip')
+    else:
+        test['requires'] = ['pip']
+
+    if 'commands' in test:
+        test['commands'].append('{{ PYTHON }} -m pip check')
+    else:
+        test['commands'] = ['{{ PYTHON }} -m pip check']
+
+
+class PipCheckMigrator(MiniMigrator):
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        """run pip check if we see python in any host sections"""
+        host_reqs_list = (
+            attrs.get("meta_yaml", {}).get("requirements", {}).get("host", [])
+        )
+        _, host_reqs = _has_python_in_host(host_reqs_list)
+
+        if "outputs" in attrs.get("meta_yaml", {}):
+            for output in attrs.get("meta_yaml", {})["outputs"]:
+                _host_reqs_list = output.get("requirements", {}).get("host", [])
+                _, _host_reqs = _has_python_in_host(_host_reqs_list)
+                host_reqs |= _host_reqs
+        return not bool(host_reqs & set(["python"]))
```

Comment:
This should check for `pip` since not all packages use `pip`.

Reply:
pip check works for packages that don't use pip too IIUIC.

https://github.com/conda-forge/conda-forge.github.io/issues/962#issuecomment-579709883

Reply:
yes, but I imagine it doesn't work if pip isn't installed.

Reply:
pip has to be in the test section, which is added by the code. It does not need to be in the host section.